### ISSUE TITLE
Porting Chalk to default Jekyll project or not?

### ISSUE THREAD

**Issue by nielsenramon** [nielsenramon/chalk#57]
I am getting a lot of issues with people not understanding how Chalk works (deploying). I could do 2 things:

1. Ditch jekyll-assets and work with a default Jekyll installation. This would mean the following:
- Have to get autoprefixer from somewhere else
- No proper asset pipeline
- No support for unsupported github pages jekyll plugins
- Ability to host the Jekyll site on the master branch instead of the output files with the deploy script

So in short option 1 would make Chalk more dummy proof but would decrease it's technical performance.

2. Keep working the same way and potentially creating a lite version without jekyll-assets and eventually create a Jekyll Theme.

What does everyone think? Would love to get some feedback.

**Comment by huyanhh** [comment]
Deploying Chalk took me a few hours mostly because I couldn't understand the instructions on the README. My case was a little different and this is only the second time I've been working with Jekyll so I expected everything to work as all the default Jekyll installations. Perhaps option 2 with maybe an FAQ or something more expansive on the original so users who don't really understand anything can know what to do?

**Comment by migueldemoura** [comment]
Static file hosting is extremely cheap or even free. Better docs are the way to go, imo.

**Comment by WendyBeth** [comment]
Hi,

I struggled to deploy Chalk, and ended up having to rearrange how I was doing things in a way that's not mentioned in your README.

I have my repository name 'WendyBeth.github.io'. In order to deploy to your user page, you *have* to use the master branch. I struggled with this quite a bit, because running the deploy script forced the deploy to gh-pages. I could change my repo's name and then see my blog, but it a) didn't port in any assets, and b) kept linking to posts as if the namespace of my blog's repo did not exist. So if I clicked on a post title from 'WendyBeth.github.io/blog', it redirected me to 'WendyBeth.github.io/posts/post-title', which of course did not exist.

What I ended up having to do was create a different default branch - I called it develop - and then change the deploy script to write everything post-processing to master. See [here](https://github.com/WendyBeth/WendyBeth.github.io/blob/develop/bin/deploy).

I don't think you need to get rid of jekyll-assets - that's actually a really cool addition to this theme and is one of the reasons (other than the *beautiful* font choices) I chose it. Fleshing out the deploy section of the README to make it explicit what to do if 1) I want to host this blog as my user page (at username.github.io) or 2) I want to host this blog on it's own page (at username.github.io/blogname) so that the posts and all internal links are aware of the namespace, would help. If there's something I completely missed, adding *that* to the README would be ideal.

All that said, I love this theme and couldn't be happier with it. 💯

**Comment by migueldemoura** [comment]
You could use Gulp to achieve something similar to the Jekyll-Assets plugin, @nielsenramon.

**Comment by ptsurbeleu** [comment]
Folks, I have been working on something alike (haven't published it yet), to be foolproof but make it easier to start with comes at an expense of flexibility (as @nielsenramin mentioned).

It could be an option to maintain two flavors - starter (get started in no time) and expert (can be bent to your will but you need to be familiar with its internals).

Humans aren't good at maintaining in the long run but automation actually excels at that, so automating it could be a way to go.

**Comment by ptsurbeleu** [comment]
UPD: Looks like I have published that work already. Here: https://rubygems.org/gems/jekyll-theme-chalk

**Comment by nielsenramon** [comment]
Decided not to port this theme to a default Jekyll project. Jekyll-assets comes with too many benefits right now. People are using Chalk because of all the custom stuff we've added anyway!

**Comment by Joli-ta** [comment]
Could you create a port to a default Pages theme, without the jekyll-assets? I like how clean it looks, and there is plenty of customization without the assets, don't you think?
